packages/client/tests/functional/batching/tests.ts

```diff
@@ -0,0 +1,75 @@
+import { faker } from '@faker-js/faker'
+
+import { waitFor } from '../_utils/tests/waitFor'
+import { NewPrismaClient } from '../_utils/types'
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+const id1 = faker.database.mongodbObjectId()
+const id2 = faker.database.mongodbObjectId()
+
+declare let newPrismaClient: NewPrismaClient<typeof PrismaClient>
+
+testMatrix.setupTestSuite(
+  () => {
+    let prisma: PrismaClient<{ log: [{ emit: 'event'; level: 'query' }] }>
+    let queriesExecuted = 0
+
+    beforeAll(() => {
+      prisma = newPrismaClient({
+        log: [{ emit: 'event', level: 'query' }],
+      })
+
+      prisma.$on('query', () => queriesExecuted++)
+    })
+
+    beforeEach(() => {
+      queriesExecuted = 0
+    })
+
+    test('batches findUnique', async () => {
+      await Promise.all([
+        prisma.user.findUnique({ where: { id: id1 } }),
+        prisma.user.findUnique({ where: { id: id2 } }),
+      ])
+
+      await waitFor(() => {
+        expect(queriesExecuted).toBe(1)
+      })
+    })
+
+    test('does not batch different models', async () => {
+      await Promise.all([
+        prisma.user.findUnique({ where: { id: id1 } }),
+        prisma.post.findUnique({ where: { id: id2 } }),
+      ])
+
+      // binary engine can retry requests sometimes, that's why we
```

Comment:
That's unexpected for me in the context of tests, but maybe I'm missing something. Do you know why it needs to retry requests?

Reply:
It would be interesting to know what happens in the query engine/db some day